docs/genes.rst

```diff
@@ -279,6 +279,34 @@ Resources for CACNA1S
 - `Annotation of CPIC Guideline for desflurane and CACNA1S, RYR1 <https://www.pharmgkb.org/chemical/PA164749136/guidelineAnnotation/PA166180457>`__
 - `CPIC® Guideline for Potent Volatile Anesthetic Agents and Succinylcholine and RYR1 and CACNA1S <https://cpicpgx.org/guidelines/cpic-guideline-for-ryr1-and-cacna1s/>`__
 
+CFTR
+=====
+
+Phenotype summary for CFTR
+-----------------------------
+
+Diplotype-phenotype mapping is used for phenotype prediction.
+
+ .. list-table::
+    :header-rows: 1
+
+    * - Phenotype
+      - Example
+    * - Increased open channel probability
+      - G551D/Reference
+    * - No change in open channel probability
+      - F508del/F508del
+    * - Significantly enhanced channel open probability
+      - S1251N/F508del
+    * - Indeterminate
+      - F508del/Reference      
```

Comment:
Yes, I agree about the use of `Favorable Response` and `Unfavorable Response. In the guideline, although G551D and other variants are both Favorable Response, they still have some difference: the Class of Recommendation of G551D is Strong, while that of other variants is Moderate. Should we keep them separate?

Reply:
@NTNguyen13, that's a good point. However, when it comes to phenotypes since PyPGx has not been tracking "classification of recommendation" for other genes, I think we should keep both of them "Favorable Response" for now to be consistent with other genes. We can always expand "Favorable Response" to two new subclasses later if necessary. But until there is a strong need, simple is always better in my mind. But if you have a strong reason for splitting "Favorable Response" into two, I'm ready to listen :)

Reply:
Hi, sorry for not replying sooner, it takes me sometimes to find and read the latest FDA documents for ivacaftor.
I used the documents from [here](https://www.accessdata.fda.gov/scripts/cder/daf/index.cfm?event=overview.process&ApplNo=207925), under Supplements section, date 12/21/2020.

- G551D mutation has strong response to ivacaftor treatment: `Improvements in percent predicted FEV1 were observed regardless of age, disease severity, sex, and geographic region. `
- Other 9 mutations (G1244E, G1349D, G178R, G551S, G970R, S1251N, S1255P, S549N, or S549R) also have strong response, but their efficacy varies: `there was a high degree of variability of efficacy responses among the 9 mutations`
- R117H mutations has weak response overall, and strong response in some studies' subgroup: `Statistically significant improvements in clinical efficacy (FEV1, CFQ-R respiratory domain) were seen in several subgroup analyses`
- Heterozygous of F508del with another responsive mutation has strong response: `in significant improvement in ppFEV1 [4.7 percent points from study baseline to average of Week 4 and Week 8 (P<0.0001)] and CFQ-R respiratory domain score [9.7 points from study baseline to average of Week 4 and Week 8 (P<0.0001)]`
- Homozygous of F508del has weak response: `CF homozygous for the F508del mutation in the CFTR gene was 1.72 percentage points (1.5% and -0.2% for patients in the KALYDECO and placebo-treated groups, respectively) and did not reach statistical significance`

Overall, they listed 97 favorable response mutations, 27 of them are either on a positive clinical response and/or in vitro experiments (Table 3 from the same document).

Based on this evidence, it's logically seems that that we could list all 39 current mutations under `Favorable response`. It may affect other downstream analysis that required the CPIC Class of recommendation, but a little post-processing doesn't hurt.

Before making the changes, what do you think about adding all 97 FDA favorable mutations into pypgx database? That would make pypgx a bit ahead of CPIC, but more up-to-date with FDA, which is the original source.

Reply:
@NTNguyen13, thanks for the information. First of all, I think it's important to establish the fact that although the works of CPIC and FDA are closely related to each other, they are two totally independent institutions where they can have significant overlaps as well as differences (here's an [example](https://pharmgkb.blogspot.com/2020/02/fda-releases-tables-on-use-of.html)). So I don't think it's necessarily correct to say FDA is "the original source" for CPIC. Of course, this doesn't mean we shouldn't prioritize FDA's data over CPIC's. That being said, why don't we first finish updating phenotype data for CFTR and then revisit the issue of adding these new alleles (with a new issue)?